### PR TITLE
ClusterMesh HA

### PR DESCRIPTION
**Pull request by thorn3r**
This adds support for running clustermesh-apiserver deployments with multiple replicas for high availability. Each clustermesh-apiserver pod runs its own etcd cluster. Depending on configuration, either the Cilium Agent or KVStoreMesh instance watches etcd in a remote cluster. All responses from the remote etcd cluster are intercepted and the header is inspected to retrieve the etcd cluster ID. If a failover event occurs and the cluster ID has changed, the remote connection is restarted to ensure that no events are missed and that no invalid data is retained. See individual commit messages for additional details.

```release-note
Add support for deploying clustermesh-apiserver with multiple replicas for high availability.
```

### CODE REVIEW

File: pkg/clustermesh/common/interceptor.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+// newUnaryInterceptor returns a new unary client interceptor that validates the
+// cluster ID of any received etcd responses.
+func newUnaryInterceptor(cl *clusterLock) grpc.UnaryClientInterceptor {
+	return func(ctx context.Context, method string, req, reply any, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+		if err := invoker(ctx, method, req, reply, cc, opts...); err != nil {
+			return err
+		}
+		resp, ok := reply.(etcdResponse)
+		if !ok {
+			cl.errors <- ErrEtcdInvalidResponse
```
Comment (giorio94) [inline pkg/clustermesh/common/interceptor.go]: Drive-by comment, all these write to channel operations should be guarded by a select statement, to avoid deadlocking after that the first error has been written, given that there would be no receiver at that point.

**Review by giorio94 — CHANGES_REQUESTED** [review]
Could you please drop the third commit (the formatting of the workflow). [It already happened once](https://github.com/cilium/cilium/pull/27193/commits/be2b725a90a674ebb498a16155d03e2ec8e8cc1d) (in the other direction), and it caused an incredible amount of issues down the line when backporting subsequent changes.

**Comment by thorn3r** [comment]
> Could you please drop the third commit (the formatting of the workflow). [It already happened once](https://github.com/cilium/cilium/pull/27193/commits/be2b725a90a674ebb498a16155d03e2ec8e8cc1d) (in the other direction), and it caused an incredible amount of issues down the line when backporting subsequent changes.

Sure, maybe I need to find a new plugin to prevent it from exploding on save :sweat_smile:

**Comment by thorn3r** [comment]
Putting back into draft, we shouldn't skip the entire statuschecker
